lib/ruby_lsp/listeners/code_lens.rb

```diff
@@ -229,7 +229,10 @@ def add_test_code_lens(node, name:, command:, kind:, id: name)
         ).returns(String)
       end
       def generate_test_command(group_stack: [], spec_name: nil, method_name: nil)
-        command = BASE_COMMAND + T.must(@path)
+        command = BASE_COMMAND
+        command += " -Itest" if T.must(@path).include?("#{File::SEPARATOR}test#{File::SEPARATOR}")
+        command += " -Ispec" if T.must(@path).include?("#{File::SEPARATOR}spec#{File::SEPARATOR}")
+        command += " #{T.must(@path)}"
```

Comment:
I think we can use `File.fnmatch?` to check if the path matches a specific glob. I'm afraid it could cause some weirdness on Windows since paths are sometimes represented with backslash or forwardslash and I'm not 100% sure which string we would get here.

```suggestion
        path = T.must(@path)
        command = BASE_COMMAND
        command += " -Itest" if File.fnmatch?("**/test/**/*", path, File::FNM_PATHNAME)
        command += " -Ispec" if File.fnmatch?("**/spec/**/*", path, File::FNM_PATHNAME)
        command += " #{path}"
```